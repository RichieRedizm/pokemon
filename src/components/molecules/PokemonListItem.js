import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const PokemonListItem = ({ pokemon: { id, name, image, number } }) => {
  return (
    <div data-testid='pokemon-item-card' className='card list-item text-center'>
      <h3>{name}</h3>
      <Link to={`/pokemon/${id}/${name}`}>
        <button className='btn btn-dark btn-sm'>stats</button>
        <div className='image-wrap'>
          <img src={image} alt='pokemon' />
        </div>
      </Link>
      <strong className='number'>{number}</strong>
    </div>
  )
}

PokemonListItem.propTypes = {
  pokemon: PropTypes.object.isRequired
}

export default PokemonListItem
