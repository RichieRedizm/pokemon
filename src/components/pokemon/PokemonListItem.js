import React from 'react'
import PropTypes from 'prop-types'
// import { Link } from 'react-router-dom'

const PokemonListItem = ({ pokemon: { name } }) => {
  return (
    <div className='card text-center pokemon-item-card'>
      {/* <img
        src={avatar_url}
        alt='profile'
        className='round-img'
        style={{ width: '60px' }}
      /> */}
      <h3>{name}</h3>
      {/* <Link to={`/user/${login}`} className='btn btn-dark btn-sm my-1'>
        more
      </Link> */}
    </div>
  )
}

PokemonListItem.propTypes = {
  pokemon: PropTypes.object.isRequired
}

export default PokemonListItem
