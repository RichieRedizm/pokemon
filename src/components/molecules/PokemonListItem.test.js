import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import PokemonListItem from './PokemonListItem'

test('it displays a pokemon list item', () => {
  const pokemon = {
    id: 'UG9rZW1vbjowMDE=',
    number: '001',
    name: 'Bulbasaur',
    image: 'https://img.pokemondb.net/artwork/bulbasaur.jpg',
    __typename: 'Pokemon'
  }
  const { getByTestId, asFragment } = render(
    <Router>
      <PokemonListItem pokemon={pokemon} />
    </Router>
  )

  expect(getByTestId('pokemon-item-card'))
  expect(asFragment()).toMatchSnapshot()
})
