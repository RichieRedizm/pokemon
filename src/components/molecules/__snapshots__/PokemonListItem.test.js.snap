// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays a pokemon list item 1`] = `
<DocumentFragment>
  <div
    class="card text-center"
    data-testid="pokemon-item-card"
  >
    <h3>
      Bulbasaur
    </h3>
    <a
      href="/pokemon/UG9rZW1vbjowMDE=/Bulbasaur"
    >
      <button
        class="btn btn-dark btn-sm"
      >
        stats
      </button>
      <img
        alt="pokemon"
        src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
      />
    </a>
    <strong
      class="number"
    >
      001
    </strong>
  </div>
</DocumentFragment>
`;
